### PR TITLE
[Model] Add gemma model support.

### PR DESCRIPTION
**Pull request by marvin-Yu**
TODO list:
- [x] [xDNN] add `gelu` ACT support.
- [x] [mlp] add `gelu` ACT support. 
- [x] [mlp] fix catup/gate gemm bug.

### CODE REVIEW

**Comment by abenmao** [comment]
Add support in example.cpp please.

**Comment by marvin-Yu** [comment]
> Add support in example.cpp please.

the Gemma support for example.cpp will be updated in a new PR, not adding to the new content of this PR.

File: src/layers/attention.h [hunk 1]
```diff
@@ -218,6 +218,7 @@ class Attention {
             bool useSelfAttn, bool doLnBefore, int *positionIds = nullptr) {
 
         auto hiddenSize = ctx->hiddenSize;
+        auto attSize = ctx->attHeadNum * ctx->attHeadSize;
```
Comment (pujiang2018) [inline src/layers/attention.h:221]: if 'attSize' not used, let's remove it.

File: src/layers/mlp_llama.h [hunk 2]
```diff
@@ -199,7 +211,16 @@ class LlamaMLP : public SingletonBase<LlamaMLP<WeiT>> {
         const float *sumB = gateWeightSum.Data();
         ImT *C = output.Data();
 
-        ctx->mmHelper->compute_silu(false, M, N, K, 1.0f, A, lda, B, scaleB, zeroB, sumB, 0.0f, C, ldc);
+        if (ctx->actType == DecoderContext::SILU) {
+            ctx->mmHelper->compute_silu(false, M, N, K, 1.0f, A, lda, B, scaleB, zeroB, sumB, 0.0f, C, ldc);
+        } else if (ctx->actType == DecoderContext::SWIGLU) { // chatglm2/3
```
Comment (pujiang2018) [inline src/layers/mlp_llama.h:216]: let's use the original path.

Comment (marvin-Yu) [reply src/layers/mlp_llama.h:216]: Done

File: src/layers/attention.h (hunk 1 above)
Comment (marvin-Yu) [reply src/layers/attention.h]: Done

**Review by pujiang2018 — APPROVED** [review]

File: src/utils/decoder_util.h [hunk 3]
```diff
@@ -453,6 +453,36 @@ class DecoderUtil {
         }
     }
 
+    // compute gelu on the left half and then add it with the right half
+    template <typename T1, typename T2>
+    static void geluSum(hpj::Matrix<T1> &src, hpj::Matrix<T2> &dst) {
+        __m512 c1 = _mm512_set1_ps(0.044715f);
```
Comment (changqi1) [inline src/utils/decoder_util.h:459]: const

**Comment by changqi1** [comment]
Wait for checking Silu API.

File: src/utils/decoder_util.h (hunk 3 above)
Comment (marvin-Yu) [reply src/utils/decoder_util.h]: Done

**Pull request merged**